src/lib/ares_process.c

```diff
@@ -890,7 +999,11 @@ ares_status_t ares__send_query(struct query *query, struct timeval *now)
 
   /* Choose the server to send the query to */
   if (channel->rotate) {
+    /* Pull random server */
     server = ares__random_server(channel);
+  } else if (channel->optmask & ARES_OPT_SERVER_FAILOVER) {
+    /* Pull server with failover behavior */
+    server = ares__failover_server(channel);
```

Comment:
I think we should probably make this always-on with some reasonable default settings that can be overwritten which would resolve #717

Reply:
Happy to make this default. I have done this with a 10% retry probability and a minimum delay of 5 seconds between retries. (The default timeout is 2 seconds and I think we want to be more than this.)

Please let me know if you'd prefer different values.